### PR TITLE
JDK-8259539: JDK-8255711 broke trap messages

### PR DESCRIPTION
**Pull request by tstuefe**
SIGTRAP/SIGILL are used on ppc/aarch64 to implement traps in compiled code (print a trap message and stop with a fatal error).

JDK-8255711 reworked and unified that coding. Calls down into VMError::report_and_die() had been factored out of platform dependent code. Since that change, traps are handled the same as normal crashes, so we loose the detail message associated with a trap. This functionality should be restored.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8259539](https://bugs.openjdk.java.net/browse/JDK-8259539): JDK-8255711 broke trap messages

### Reviewers
 * [Lois Foltan](https://openjdk.java.net/census#lfoltan) (@lfoltan - **Reviewer**) ⚠️ Review applies to daf7dc500e1475d874c528ae369470480fbc0375
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2022/head:pull/2022`
`$ git checkout pull/2022`

### CODE REVIEW

File: src/hotspot/os_cpu/aix_ppc/os_aix_ppc.cpp [hunk 1]
```diff
@@ -314,6 +314,14 @@ bool PosixSignals::pd_hotspot_signal_handler(int sig, siginfo_t* info,
...
 
+        // End life with a fatal error, message and detail message and the context.
+        // Note: no need to do any post-processing here (e.g. signal chaining)
+        va_list va_dummy;
+        VMError::report_and_die(thread, uc, NULL, 0, msg, detail_msg, va_dummy);
+        va_end(va_dummy);
+
+        ShouldNotReachHere();
+
         return false; // Fatal error
```
Comment (lfoltan) [inline src/hotspot/os_cpu/aix_ppc/os_aix_ppc.cpp:325]: In the two files following on you removed the return statement.  For consistency can you do the same here?

**Review by lfoltan — APPROVED** [review]
LGTM with one suggestion.
Lois

File: src/hotspot/os_cpu/aix_ppc/os_aix_ppc.cpp (hunk 1 above)
Comment (tstuefe) [reply src/hotspot/os_cpu/aix_ppc/os_aix_ppc.cpp:325]: Oh, you are right. Will do. 

Thanks, Lois!

**Review by dholmes-ora — APPROVED** [review]
LGTM.

Thanks,
David

**Pull request closed without merging**